redis/config.go

```diff
@@ -0,0 +1,36 @@
+package redis
+
+import (
+	"github.com/HomesNZ/go-common/env"
+	validation "github.com/go-ozzo/ozzo-validation"
+)
+
+type DeclareConfig func(config *Config)
+
+type Config struct {
```

Comment:
make it interface as well, and no need to expose Host and Port, you can make the address function instead.